racket/src/ChezScheme/c/version.h

```diff
@@ -91,7 +91,9 @@ FORCEINLINE void store_unaligned_uptr(uptr *addr, uptr val) {
 #define GETPAGESIZE() getpagesize()
 typedef char *memcpy_t;
 #define MAKE_NAN(x) { x = 0.0; x = x / x; }
-#define GETWD(x) getcwd((x),PATH_MAX)
+#ifdef PATH_MAX
+# define GETWD(x) getcwd((x),PATH_MAX)
+#endif
 typedef int tputsputcchar;
```

Comment:
This caused a problem that wasn't exposed until I tried to cross-compile Racket CS using Guix (this patch had fixed only Chez Scheme). Apparently we have not yet included `<limits.h>` at this point, so `PATH_MAX` is never defined here: all of the references on the right-hand sides of `#define` are unhygienic. The problem was masked by the fallback implementation of `current-directory` using `/bin/pwd`:
https://github.com/racket/racket/blob/2f84598893f3409fee9f2b0d5c952cf09be98916/racket/src/ChezScheme/s/6.ss#L37-L50
Guix revealed the issue because `/bin` doesn't exist in its build environment. It seems to have been just coincidental that the failure came up while cross-compiling specifically: apparently most build modes don't exercise `(current-directory)`, and I have the Chez test suite disabled again in Guix due to https://github.com/racket/racket/issues/4359#issuecomment-1202036929.

In some sense it seems like the "right thing" would be to include `<limits.h>` or test in `configure` or something, but I'm inclined to change the test to `#ifndef __GNU__`, instead: that's the style of the other conditionals here, and it should keep the change small and prevent any other surprises for non-Hurd systems.

It might be more robust to remove this use of `PATH_MAX` across the board along the lines of https://github.com/racket/racket/commit/b4d452cc71c1ca6f848d7404c09d75f14f51d170—in this case, we're dynamically allocating the storage anyway!—but that seems like a bigger change that should go in its own patch.

Reply:
I've pushed a revised commit, which I'm now testing.